### PR TITLE
Add model metadata to request schema

### PR DESCRIPTION
**Pull request by AlyssaCote**
This PR adds a `Model` schema in the `Request` schema with metadata about the model included so that it's not just a data blob. I also tweaked some of the attribute names. For example, what was `request.input.inputKeys` is now `request.input.keys`.

### CODE REVIEW

File: smartsim/_core/mli/infrastructure/control/workermanager.py [hunk 1]
```diff
@@ -65,12 +66,12 @@ def deserialize_message(
...
     # return request
     model_key: t.Optional[str] = None
-    model_bytes: t.Optional[bytes] = None
+    model_bytes: t.Optional[Model] = None
 
-    if request.model.which() == "modelKey":
-        model_key = request.model.modelKey.key
-    elif request.model.which() == "modelData":
-        model_bytes = request.model.modelData
+    if request.model.which() == "key":
```
Comment (ankona) [inline smartsim/_core/mli/infrastructure/control/workermanager.py:71]: ❤️ loving the simplified naming

File: smartsim/_core/mli/mli_schemas/model/model.capnp [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+@0xaefb9301e14ba4bd;
+
+struct Model {
+  data @0 :Data;
+  name @1 :Text;
+  version @2 :Text;
+}
```
Comment (ankona) [inline smartsim/_core/mli/mli_schemas/model/model.capnp:33]: trailing newline to make linters happy?

Comment (AlyssaCote) [reply smartsim/_core/mli/mli_schemas/model/model.capnp]: Added a newline to all `capnp` schema files!

File: smartsim/_core/mli/mli_schemas/model/model_capnp.pyi [hunk 3]
```diff
@@ -0,0 +1,46 @@
+"""This is an automatically generated stub for `model.capnp`."""
```
Comment (ankona) [inline smartsim/_core/mli/mli_schemas/model/model_capnp.pyi:1]: BSD license

File: smartsim/_core/mli/mli_schemas/model/model_capnp.py [hunk 4]
```diff
@@ -0,0 +1,12 @@
+"""This is an automatically generated stub for `model.capnp`."""
```
Comment (ankona) [inline smartsim/_core/mli/mli_schemas/model/model_capnp.py:1]: license

**Review by ankona — CHANGES_REQUESTED** [review]
I see a couple minor license issues and a potential breaking bug to fix. Otherwise, looks good.

File: smartsim/_core/mli/infrastructure/control/workermanager.py [hunk 5]
```diff
@@ -91,19 +92,19 @@ def deserialize_message(
...
+        input_meta = [data.tensorDescriptor for data in request.input.data]
 
     inference_request = InferenceRequest(
         model_key=model_key,
         callback=comm_channel,
         raw_inputs=input_bytes,
         input_meta=input_meta,
         input_keys=input_keys,
-        raw_model=model_bytes,
+        raw_model=model_bytes.data if model_bytes is not None else None,
```
[the remark this answers is not in the document]
Comment (AlyssaCote) [reply smartsim/_core/mli/infrastructure/control/workermanager.py:107]: The new `Model` schema has a `data` attribute that holds the bytes. `request.model.data` accesses the `Model` object, and the one more `.data` accesses the bytes within the model object. If we just want to pass around the raw bytes, then I think this is correct, but if we want to pass the `Model` schema object around (which I think we do, now that I'm seeing it), then I think some renaming may need to happen?

[the remark this answers is not in the document]
Comment (AlyssaCote) [reply smartsim/_core/mli/infrastructure/control/workermanager.py]: What I've done is pass in `model_bytes` which is now a `Model` schema object, and then later on when bytes need to be passed into `FetchModelResults` I pass in the `data` attribute, which is bytes. Let me know what you think!

[the remark this answers is not in the document]
Comment (ankona) [reply smartsim/_core/mli/infrastructure/control/workermanager.py]: that makes sense, now. it looked like repeating the same method call but was not. resolving!

**Review by ankona — APPROVED** [review]
LGTM

**Pull request merged**